consensus/hotstuff/notifications/slashing_violation_consumer.go

```diff
@@ -23,16 +24,17 @@ func NewSlashingViolationsConsumer(log zerolog.Logger) *SlashingViolationsConsum
 		log: log,
 	}
 }
-func (c *SlashingViolationsConsumer) OnInvalidBlockDetected(err model.InvalidProposalError) {
-	block := err.InvalidProposal.Block
+func (c *SlashingViolationsConsumer) OnInvalidBlockDetected(err flow.Slashable[model.InvalidProposalError]) {
```

Comment:
I feel like this use of `Slashable` is inconsistent with it's documentation:

> // Slashable represents an message we got from a different node, whose validity has not been _entirely_
> // confirmed.

Initially, we intended to use `Slashable` to wrap messages which haven't been validated yet. But here we're wrapping an error type, which contains a message which _has_ been validated (but is invalid). 

I'm wondering if we should create a new type which represents a message for which we _intend_ to submit slashing evidence:
- `flow.Slashable` wraps a message with information we _could_ use to slash the sender, if the message turns out to be invalid
- `flow.SlashingViolation` wraps a slashable message and context about a slashable violation
```go
type SlashingViolation[Err error, Msg any] struct {
  Message Slashable[Msg]
  Err Err
}
```

Reply:
The reason I have used it this way is because `Slashable` is still in framing process, I don't really know what will be the final design. For now I think it's ok to use `Slashable` as construction which just attaches `OriginID` to _any_ message without extra limiting in which areas it can be used. Once we are working on slashing, especially generating and submitting proofs it will be a lot easier to properly design it. I would suggest to leave all revisions for future.